src/planner/binder/expression/bind_window_expression.cpp

```diff
@@ -139,6 +139,12 @@ BindResult BaseSelectBinder::BindWindow(WindowExpression &window, idx_t depth) {
 	}
 	for (auto &order : window.orders) {
 		BindChild(order.expression, depth, error);
+		if (error.empty()) {
+			//	Restore any collation expressions
+			auto &order_expr = order.expression;
+			auto &bound_order = BoundExpression::GetExpression(*order_expr);
+			ExpressionBinder::PushCollation(context, bound_order, bound_order->return_type, false);
```

Comment:
Can this be done instead after the `if (!error.empty())` on line 156?

I'm not sure if this matters but in the current state this code can be executed multiple times in case of correlated subqueries where we have stacked binders.